widget/numerical_entry.go

```diff
@@ -33,6 +35,11 @@ func (e *NumericalEntry) TypedRune(r rune) {
 
 	if e.AllowFloat && (r == '.' || r == ',') {
 		e.Entry.TypedRune(r)
+		return
+	}
+
+	if r == '-' && e.AllowNegative && e.Entry.CursorRow == 0 && e.Entry.CursorColumn == 0 {
```

Comment:
Thinking about this, I wonder if we need to be smarter about how we allow writing negative signs. Right now you will be able to write multiple negative signs as long as you are at the first position. Is that something we should handle?

Reply:
I have updated the code to not allow input in front of '-' character.

Reply:
By the way, multiple '.' and ',' are allowed. I have a fix for that as well. Will do a PR after this one is accepted.

Reply:
> By the way, multiple '.' and ',' are allowed. I have a fix for that as well. Will do a PR after this one is accepted.

Cool. You can put dots or commas to make large numbers more readable. I think the solution for that is a bit complicated.

Reply:
Actually not very complicated. But should multiple dot/commas be allowed? If yes, then I will throw my "fix" away.

Reply:
I suppose the answer depends on what the strconv.ParseFloat function allows. We should likely match that